pkg/metrics/otelmetrics/factory.go

```diff
@@ -0,0 +1,32 @@
+package otelmetrics
+
+import "github.com/jaegertracing/jaeger/pkg/metrics"
+
+type otelFactory struct{}
+
+func NewOTELFactory() metrics.Factory {
+	return &otelFactory{}
+}
+
+func (f *otelFactory) Counter(opts metrics.Options) metrics.Counter {
+	return newOtelCounter(opts.Name, opts.Tags)
```

Comment:
There is no value in calling additional function here. Why not implement logic right here?

Reply:
I was thinking that writing the function for each metric in a separate file instead of cramming it in factory.go would be cleaner?

Reply:
Factory file is tiny. It's not "cleaner" to force the reader to go to another file, nor is it efficient to have another indirection.

Reply:
Noted, I will initialize the struct and respective function in the separate file and initialize the counter in the factory file itself.